### PR TITLE
topology1: CMakeLists: add RPLP CRB support for SoundWire

### PR DESCRIPTION
**Pull request by apoorvintel**
Added RPLP CRB support for SoundWire0 and SoundWire2

@bardliao  , @CBala21 , Please review it !

### CODE REVIEW

**Comment by sofci** [comment]
Can one of the admins verify this patch?
> reply **test&nbsp;this&nbsp;please** to run this test once

File: tools/topology/topology1/CMakeLists.txt [hunk 1]
```diff
@@ -51,6 +51,8 @@ set(TPLGS
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l0-rt1316-l13-rt714-l2\;-DPLATFORM=adl\;-DUAJ_LINK=0\;-DAMP_1_LINK=1\;-DAMP_2_LINK=3\;-DEXT_AMP_REF\;-DMIC_LINK=2"
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l2-rt1316-l01-rt714-l3\;-DPLATFORM=adl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DMIC_LINK=3"
 	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-adl-rt711-l2-rt1316-l01\;-DPLATFORM=adl\;-DUAJ_LINK=2\;-DAMP_1_LINK=0\;-DAMP_2_LINK=1\;-DEXT_AMP_REF\;-DNO_LOCAL_MIC"
+	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-rpl-rt711-l2\;-DPLATFORM=rpl\;-DUAJ_LINK=2\;-DNOAMP\;-DNO_LOCAL_MIC"
+	"sof-icl-rt711-rt1308-rt715-hdmi\;sof-rpl-rt711-l0\;-DPLATFORM=rpl\;-DUAJ_LINK=0\;-DNOAMP\;-DNO_LOCAL_MIC"
```
Comment (plbossart) [inline tools/topology/topology1/CMakeLists.txt:55]: @bardliao this is inconsistent with the addition of rpl support line 69 https://github.com/thesofproject/sof/pull/7555/commits/2368af419d99484c1f29965784784b5428c5d6cd#diff-8b3b8af0931428ddcd505402f24cd6e1d1d84151c6c16f8152ef96a99c659a82R69

something's not quite right here

Comment (bardliao) [reply tools/topology/topology1/CMakeLists.txt:55]: > @bardliao this is inconsistent with the addition of rpl support line 69 [2368af4#diff-8b3b8af0931428ddcd505402f24cd6e1d1d84151c6c16f8152ef96a99c659a82R69](https://github.com/thesofproject/sof/commit/2368af419d99484c1f29965784784b5428c5d6cd#diff-8b3b8af0931428ddcd505402f24cd6e1d1d84151c6c16f8152ef96a99c659a82R69)
> 
> something's not quite right here

"sof-tgl-rt711-rt1308\;sof-rpl-rt711\;-DCHANNELS=0\;-DBT_OFFLOAD\;-DPLATFORM=rpl" looks the same as "sof-icl-rt711-rt1308-rt715-hdmi\;sof-rpl-rt711-l0\;-DPLATFORM=rpl\;-DUAJ_LINK=0\;-DNOAMP\;-DNO_LOCAL_MIC" except the naming.
Consider
[.sof_tplg_filename = "sof-rpl-rt711-l0.tplg",](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-acpi-intel-rpl-match.c#L372)
[.sof_tplg_filename = "sof-rpl-rt711-l2.tplg",](https://github.com/thesofproject/linux/blob/topic/sof-dev/sound/soc/intel/common/soc-acpi-intel-rpl-match.c#L378)
Maybe we don't need "sof-tgl-rt711-rt1308\;sof-rpl-rt711\;-DCHANNELS=0\;-DBT_OFFLOAD\;-DPLATFORM=rpl"?

Comment (plbossart) [reply tools/topology/topology1/CMakeLists.txt:55]: I don't know how RPL support was added @bardliao , this doesn't seem quite right to me to use different topology files to generate the same jack-only configuration and have different naming conventions.

Comment (bardliao) [reply tools/topology/topology1/CMakeLists.txt:55]: > I don't know how RPL support was added @bardliao , this doesn't seem quite right to me to use different topology files to generate the same jack-only configuration and have different naming conventions.

I would suggest removing "sof-tgl-rt711-rt1308\;sof-rpl-rt711\;-DCHANNELS=0\;-DBT_OFFLOAD\;-DPLATFORM=rpl". What do you think? @plbossart

Comment (CBala21) [reply tools/topology/topology1/CMakeLists.txt:55]: Hi @bardliao ,

I think this file name(sof-rpl-rt711) has been used for kernel 5.19 above where rpl_desc and match table has been introduced and used. The removal may impact the latest kernel version users else driver need to be changed in soc-apci-intel-rpl-match.c to use the updated topology filename(sof-rpl-rt711-l0)

Comment (plbossart) [reply tools/topology/topology1/CMakeLists.txt:55]: yes, it's a problem to change now. Probably we need to keep this and move on.

**Review by plbossart — APPROVED** [review]

**Review by bardliao — APPROVED** [review]

**Review by CBala21 — APPROVED** [review]

**Comment by kv2019i** [comment]
Ready to go, needs a clean CI pass.

**Comment by plbossart** [comment]
CI is completely stuck @lgirdwood @kv2019i

**Comment by kv2019i** [comment]
@wszypelt  can you check the quickbuild errors?

**Comment by kv2019i** [comment]
SOFCI TEST

**Comment by kv2019i** [comment]
Thanks all, merging. Please send backport PR to stable-v2.2.

**Pull request merged**